### ISSUE TITLE
test files create parser issue

### ISSUE THREAD

**Issue by EdJoPaTo** [xojs/eslint-config-xo-typescript#15]
As one example of many (one per file within `test/`):
```
test/math/probability.ts
✖    0:0    Parsing error: If "parserOptions.project" has been set for @typescript-eslint/parser, …/test/math/probability.ts must be included in at least one of the projects provided.
```

**Comment by sindresorhus** [comment]
`parserOptions.project` has to be set for some of the rules to work. If we didn't set it here, you would have to set it yourself, and the result would be the same.

See https://github.com/typescript-eslint/typescript-eslint/issues/890

**Comment by EdJoPaTo** [comment]
As xo also includes rules for Ava I would assume it works out of the box with ava tests. Which is the purpose of xo: work with as little config as possible.

**Comment by fregante** [comment]
@sindresorhus why is this even a problem with `xo`? `eslint` doesn't require me to mess with `tsconfig` to enable linting. I could run `eslint *.ts` and it would work as expected.

**Comment by fregante** [comment]
The solution suggested in https://github.com/typescript-eslint/typescript-eslint/issues/890 silences the error but `xo` doesn't report the errors in the second file. I think it's because `tsconfig`'s `extend` is buggy.

```js
// tsconfig.json
{
	"extends": "@sindresorhus/tsconfig",
	"compilerOptions": {
		"outDir": ".",
		"target": "es2018",
		"module": "es2015"
	},
	"files": [
		"index.ts"
	]
}
```
```js
// tsconfig.xo.json
{
	"extends": "./tsconfig.json",
	"files": [
		"index.ts",
		"index.test-d.ts"
	]
}
```
```js
// package.json
{
  "xo": {
    "extensions": [
      "ts"
    ],
    "extends": "xo-typescript",
    "parserOptions": {
       "project": "./tsconfig.xo.json"
    }
  }
}
```

**Comment by sindresorhus** [comment]
@fregante This is not an XO issue. The problem is in TypeScript-ESLint. I'm happy to consider ways XO to simplify this though.

**Comment by fregante** [comment]
Yeah this is an annoying requirements of ts-eslint.

> I'm happy to consider ways XO to simplify this though.

XO could:

- avoid linting TS files not included in tsconfig (not great)
- detect the error and suggest the above solution, if possible
- since `xo` is setting the `project` prop, perhaps it can also generate/extend one on the fly and include it in there, but it's probably hard to accomplish.

I don't know of a better solution to the problem in the first place
